src/smc_pyutil/smc_pyutil/smc_open.py

```diff
@@ -50,7 +50,8 @@ def main():
         print "Opens each file (or directory) in the Sagemath Cloud web-based editor from the shell."
         print "If the named file doesn't exist, it is created."
     else:
-        process(sys.argv[1:])
+        # ignore args that have ? or * after glob expansion
+        process([p for p in sys.argv[1:] if re.search('[\?\*]', p) is None])
```

Comment:
Harald has a good point about other shell expansions, so that needs to be looked up.

Also, using a regular expression not only makes I think the code harder to read and work on, it turns out it is also much, **much** slower:  https://cloud.sagemath.com/projects/4a5f0542-5873-4eed-a85c-a18c706e8bcd/files/support/2016-10-22-issue1019.sagews

Of course the speed doesn't matter -- in both cases (with or without re) it is fast.  However, regular expressions are often notoriously hard to read and lead to weird bugs, whereas doing, e.g., `'?' in p` is very clear and easy to read code... and there is no loss in speed (in fact, it is over 10x faster).